tests/Models/TokenSigningTraitTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * @author Chris Chabilall <chris.c@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license http://www.opensource.org/licenses/gpl-2.0.php GNU GPL v2
+ */
+
+namespace VanillaTests\Models;
+
+use VanillaTests\SharedBootstrapTestCase;
+use VanillaTests\SiteTestTrait;
+use VanillaTests\Fixtures\TokenTestingModel;
+
+/**
+ * Test the {@link TokenSigningTrait}.
+ * Used to test the token generation and signing utility methods.
+ */
+class TokenSigningTraitTest extends SharedBootstrapTestCase {
+    use SiteTestTrait;
+
+    /**
+     * Tests random token generation and signing.
+     */
+    public function testVerifyRandomTokenSignature() {
+        $model = new TokenTestingModel();
+        $token = $model->signToken($model->randomToken(), '2 months');
+        $this->assertEquals(true, $model->verifyTokenSignature($token, true));
+    }
+
+    /**
+     * An expired token shouldn't verify.
+     *
+     * @expectedException \Exception
+     * @expectedExceptionMessage Your nonce has expired.
+     */
+    public function testExpiryDate() {
+        $model = new TokenTestingModel();
+        $token = $model->signToken($model->randomToken(), 'last year');
+        $model->verifyTokenSignature($token, true);
+    }
+
+    /**
+     * An altered token signature shouldn't verify.
+     *
+     * @expectedException \Exception
+     * @expectedExceptionMessage Your nonce has an invalid signature.
+     */
+    public function testBadSignature() {
+        $model = new TokenTestingModel();
+        $token = $model->signToken($model->randomToken(), '2 months').'!';
+        $model->verifyTokenSignature($token, true);
+    }
+
+    /**
+     * A nonsense token shouldn't verify.
+     *
+     * @expectedException \Exception
+     * @expectedExceptionMessage nonce missing parts.
```

Comment:
I'm concerned this exception message isn't correct and the test could still be passing.

Reply:
for some reason the test fails when it should